### PR TITLE
Moved CutePDF to AU.

### PR DESCRIPTION
**Pull request by mikecole**

### CODE REVIEW

File: automatic/PDFCreator/PDFCreator.nuspec [hunk 1]
```diff
@@ -17,8 +17,4 @@ PDFCreator – PDFCreator easily creates PDFs from any Windows program. Use it l
     <tags>pdf admin</tags>
     <packageSourceUrl>https://github.com/chocolatey/chocolatey-coreteampackages</packageSourceUrl>
```
Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: https://github.com/chocolatey/chocolatey-coreteampackages/tree/master/automatic/PDFCreator

Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: Its also good chance (although not mandatory) to update other metadata:
- Give some features
- Add some more tags such as `editor`, `office` etc.
- Add docs, bugtrackerurl, mailinglist etc.

Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: Also, its open source so, `projectsSourceUrl`: https://github.com/pdfforge/PDFCreator

File: automatic/PDFCreator/update.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     }
+}
+
+function global:au_GetLatest {
+    Write-Host $url
+
+    $temp_file = $env:TEMP + '\PDFCreator.exe'
+    Invoke-WebRequest $url -OutFile $temp_file
+
+    $version = (Get-Command $temp_file).Version
```
Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: Interesting way to obtain the version. I like it.

However, its not appropriate here. The file size is ~27MB and the version is available on download page.

So you should find out something that produces faster update:
- Download page has info about the version.
- Blog seems to serve as release new information: http://www.pdfforge.org/blog, so you can take in their RSS feed (http://www.pdfforge.org/blog/rss) and filter items with PDFCreator in title and get the version.

**Comment by majkinetor** [comment]
In advance, please use separate PR for each package,  please.

File: automatic/PDFCreator/PDFCreator.nuspec (hunk 1 above)
Comment (mikecole) [inline automatic/PDFCreator/PDFCreator.nuspec]: I was going more for a 1-1 conversion from ketarin to AU, but I'll add these.

File: automatic/PDFCreator/update.ps1 (hunk 2 above)
Comment (mikecole) [inline automatic/PDFCreator/update.ps1]: The download page gives me 2.4. The blog gives me 2.4.0. Getting the version from the installer gives me 2.4.0.213. I wasn't sure what level of detail was most appropriate, so I went with most detailed. What is your opinion on which version is appropriate?

**Comment by mikecole** [comment]
Sorry, didn't realize two packages snuck into this pull request.

File: automatic/PDFCreator/update.ps1 (hunk 2 above)
Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: Its a good find. I had a same problem with LibreOffice - on site it was one version and it adds revision number. It was important to recognize what you had in order not to download if not needed. However, you solution downloads too, so it doesn't make a difference. But its great soluion when its very hard to parse the version or if its impossible.

Just go with the site 2.4 or blog 2.4.0 (site is probably what is expected by users tbh). Its better not to add revision as it can be used for choco fix.

Comment (gep13) [inline automatic/PDFCreator/update.ps1]: I would go with 2.4.0, I.e. What comes from the blog post.

File: automatic/PDFCreator/PDFCreator.nuspec (hunk 1 above)
Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: This migration to AU is perfect time to tidy up metadata. You need to revisit everything anyway, and those information are really handy to have in `choco info <package>` or on site directly.

File: automatic/PDFCreator/update.ps1 (hunk 2 above)
Comment (mikecole) [inline automatic/PDFCreator/update.ps1]: OK, I can look at the RSS feed for the first title that contains "released" and extract the version. It might not be perfect, but it's probably good enough.

File: automatic/PDFCreator/PDFCreator.nuspec (hunk 1 above)
Comment (mikecole) [inline automatic/PDFCreator/PDFCreator.nuspec]: 👍

File: automatic/PDFCreator/update.ps1 (hunk 2 above)
Comment (mikecole) [inline automatic/PDFCreator/update.ps1]: Yeah that's probably the right choice.

Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: I wouldnt add 0 just like that. What happens when they tomorrow release 2.5.0.1. You would get invalid version. I think u should use site version.

2.5 is not equal to 2.5.0 BTW.

Comment (gep13) [inline automatic/PDFCreator/update.ps1]: @majkinetor ok, I have to ask, why is 2.5 not the same as 2.5.0?  In one, the patch version is explicitly being stated as 0, and the other, it just isn't being shown.

The suggestion wasn't to simply add a .0, but rather, if the version number is a 2 part version number, add the .0 onto it.

Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: > The suggestion wasn't to simply add a .0, but rather, if the version number is a 2 part version number, add the .0 onto it.

I think this is irrelevant. 

> ok, I have to ask, why is 2.5 not the same as 2.5.0? 

Because `[version]2.5 -ne [version]2.5.0` :)

**Comment by majkinetor** [comment]
@mikecole  I referenced few issues reported for those packages. Would u please take a look?

**Comment by mikecole** [comment]
Sure will. Thanks!

File: automatic/PDFCreator/update.ps1 (hunk 2 above)
Comment (gep13) [inline automatic/PDFCreator/update.ps1]: @majkinetor ah, I see the problem here... We are not talking about the same thing.  I was talking about Semantic Versioning.

image

Comment (AdmiringWorm) [inline automatic/PDFCreator/update.ps1]: I see the PDF creator uses minimum tree part version, probably best to follow the existing convention.
https://chocolatey.org/packages/PDFCreator

Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: That makes sense.

**Comment by mikecole** [comment]
@majkinetor I believe I fixed both linked issues.

File: automatic/PDFCreator/update.ps1 [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import-module au
+
+$download_page_url = 'http://download.pdfforge.org/download/pdfcreator/list'
+
+function global:au_SearchReplace {
+    @{
+        'tools\ChocolateyInstall.ps1' = @{
+            "(^[$]checksum\s*=\s*)('.*')"   = "`$1'$($Latest.Checksum32)'"
```
Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: You don't have an URL here. It will never update it.

File: automatic/cutepdf/update.ps1 [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        'tools\ChocolateyInstall.ps1' = @{
+            "(^[$]checksum\s*=\s*)('.*')"   = "`$1'$($Latest.Checksum32)'"
+        }
+     }
+}
+
+function global:au_GetLatest {
+    Write-Host $url
+
+    $temp_file = $env:TEMP + '\CuteWriter.exe'
```
Comment (majkinetor) [inline automatic/cutepdf/update.ps1]: The same as before here, without downloading the files if other things are possible.

**Comment by mikecole** [comment]
@majkinetor fixed and fixed.

File: automatic/PDFCreator/PDFCreator.nuspec [hunk 5]
```diff
@@ -14,11 +14,9 @@
 PDFCreator – PDFCreator easily creates PDFs from any Windows program. Use it like a printer in Word, StarCalc or any other Windows application.
     </description>
     <summary>PDFCreator - PDFCreator easily creates PDFs from any Windows program.</summary>
-    <tags>pdf admin</tags>
-    <packageSourceUrl>https://github.com/chocolatey/chocolatey-coreteampackages</packageSourceUrl>
+    <tags>pdf editor printer admin</tags>
```
Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: Should be tagged with [fos](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guideliness-%5BDraft%5D#122-obligatory-tags).

File: automatic/PDFCreator/update.ps1 [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    $version = $newest_release.innerText.Replace("PDFCreator ", "").Trim()
+
+    $newest_release_url = $html.body.getElementsByTagName("a") | where {$_.href -like "*/download/pdfcreator/$version*"}
+
+    $url = $newest_release_url.href.Replace("about:/download", "http://orange.download.pdfforge.org") + "&download"
+
+    return @{ URL = $url; Version = $version }
+}
+
+update -NoCheckUrl -ChecksumFor 32
```
Comment (majkinetor) [inline automatic/PDFCreator/update.ps1]: AU complains about URLs ? hence `NoCheckUrl` ?

File: automatic/cutepdf/cutepdf.nuspec [hunk 7]
```diff
@@ -1,10 +1,9 @@
...
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
   <metadata>
-    <id>{{PackageName}}</id>
+    <id>cutepdf</id>
     <title>CutePDF</title>
-    <version>{{PackageVersion}}</version>
+    <version>3.0</version>
     <authors>Acro Software Inc.</authors>
     <owners>Redsandro</owners>
     <summary>Free PDF printer driver with no watermarks.</summary>
```
Comment (majkinetor) [inline automatic/cutepdf/cutepdf.nuspec]: Please add '##Features'

File: automatic/cutepdf/cutepdf.nuspec [hunk 8]
```diff
@@ -1,10 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
```
Comment (majkinetor) [inline automatic/cutepdf/cutepdf.nuspec]: Add `freeware` among tags.

**Comment by majkinetor** [comment]
Also, add releaseNotes. I see that CutePDF has them at http://www.cutepdf.com/support/updates.asp and the other package certainly has it.

Those are all metadata problems. When you fix it I will merge it.

File: automatic/PDFCreator/PDFCreator.nuspec (hunk 5 above)
Comment (AdmiringWorm) [inline automatic/PDFCreator/PDFCreator.nuspec]: that would actually be `foss`, `fos` means something completely different

Comment (majkinetor) [inline automatic/PDFCreator/PDFCreator.nuspec]: Yeah, my bad.. foss

**Comment by gep13** [comment]
@mikecole are you in a position to update this PR with the suggestions, or are you happy for us to make the modifications and merge?

File: automatic/PDFCreator/update.ps1 (hunk 6 above)
Comment (mikecole) [inline automatic/PDFCreator/update.ps1]: No, that was copied and pasted from another package. I'll fix it.

**Comment by mikecole** [comment]
@gep13 yes I can fix these in the next few days, unless you'd like them sooner.
